packages/supi/src/api/storeAdd.ts

```diff
@@ -0,0 +1,59 @@
+import normalizeRegistryUrl = require('normalize-registry-url')
+
+import {streamParser} from '@pnpm/logger'
+import {StoreController} from 'package-store'
+
+import parseWantedDependencies from '../parseWantedDependencies';
+import {ReporterFunction} from '../types'
+
+export default async function (
+  fuzzyDeps: string[],
+  opts: {
+    prefix: string,
+    registry?: string,
+    verifyStoreIntegrity: boolean,
+    reporter?: ReporterFunction,
+    storeController: StoreController,
+  },
+) {
+  const reporter = opts && opts.reporter
+  if (reporter) {
+    streamParser.on('data', reporter)
+  }
+
+  const deps = await parseWantedDependencies(fuzzyDeps, {
+    allowNew: true,
+    currentPrefs: {},
+    defaultTag: '',
+    dev: false,
+    devDependencies: {},
+    optional: false,
+    optionalDependencies: {},
+  })
+
+  const pkgIds = []
+  for (const dep of deps) {
+    const ret = await opts.storeController.requestPackage(dep, {
+      downloadPriority: 1,
+      loggedPkg: {
+        rawSpec: dep.raw,
+      },
+      preferredVersions: {},
+      prefix: opts.prefix,
+      registry: normalizeRegistryUrl(opts.registry || 'https://registry.npmjs.org/'),
+      verifyStoreIntegrity: opts.verifyStoreIntegrity,
+    })
+    pkgIds.push(ret.body.id)
+  }
+
+  await opts.storeController.updateConnections(opts.prefix, {
```

Comment:
As far as I remember, this is not the desired behavior because you don't want this prefix to be added to `store.json`

I think what we can do is, make package requester add the new fields to `storeIndex` right after downloading the packages to the store.

Then calling updateConnections won't be needed

Reply:
As you can see, storeIndex is passed in to package-requester
https://github.com/pnpm/package-store/blob/master/src/storeController/index.ts#L55

so you can just add the field in package-requester like it is done in package-store currently (but just add an empty array)

https://github.com/pnpm/package-store/blob/master/src/storeController/index.ts#L91

Reply:
I am not sure what you mean with this